### PR TITLE
ci: enable coverage

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Report Test Coverage in the CI

## Why is it important?

Help to visualise when the Test coverage gets increased or decreased (see https://github.com/elastic/elastic-agent/pull/377#issuecomment-1110031830)

## Questions

Current implementation does run coverage for each platform (windows, linux and ARM), therefore the coverage results are not aggregated, I've seen https://github.com/elastic/beats/blob/main/dev-tools/aggregate_coverage.py.

1) Is it required to aggregate the results from all platforms?
2) Does the existing coverage in https://fleet-ci.elastic.co/job/elastic-agent/job/elastic-agent-mbp/job/PR-377/4/coverage/ reflect the real test coverage?

image

## Tasks

- [x] Add cov to xml transformation in the Magefile
- [ ] Converge coverage results? It seems https://github.com/elastic/beats/blob/main/dev-tools/aggregate_coverage.py was in place but was not migrated to this project.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -266,6 +266,7 @@ github.com/bmizerany/pat v0.0.0-20170815010413-6226ea591a40/go.mod h1:8rLXio+Wji
 github.com/boltdb/bolt v1.3.1/go.mod h1:clJnj/oiGkjum5o1McbSZDSLxVThjynRyGBgiAx27Ps=
 github.com/bonitoo-io/go-sql-bigquery v0.3.4-1.4.0/go.mod h1:J4Y6YJm0qTWB9aFziB7cPeSyc6dOZFyJdteSeybVpXQ=
 github.com/boumenot/gocover-cobertura v1.2.0/go.mod h1:fz7ly8dslE42VRR5ZWLt2OHGDHjkTiA2oNvKgJEjLT0=
+github.com/boumenot/gocover-cobertura v1.2.0/go.mod h1:fz7ly8dslE42VRR5ZWLt2OHGDHjkTiA2oNvKgJEjLT0=
```
[the remark this answers is not in the document]
Comment (mdelapenya) [reply go.sum:269]: https://github.com/elastic/elastic-agent/blob/main/tools/tools.go

[the remark this answers is not in the document]
Comment (mdelapenya) [reply go.sum:269]: Btw, lines 268 and 269 are exactly the same. Can you run go mod tidy and verify that?

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -111,6 +112,7 @@ pipeline {
             post {
               always {
                 junit(allowEmptyResults: true, keepLongStdio: true, testResults: "${BASE_DIR}/build/TEST-*.xml")
+                coverageReport(baseDir: "**/build", reportFiles: 'TEST-go-unit.html', coverageFiles: 'TEST-go-unit-cov.xml')
```
Comment (v1v) [inline .ci/Jenkinsfile:115]: Question: is it required to aggregate the results from all the different platforms (windows, linux, arm)? If so, https://github.com/elastic/beats/blob/main/dev-tools/aggregate_coverage.py is the one in charge?

**Review by jlind23 — COMMENTED** [review]
Having only the Linux coverage can be a good first step.

**Comment by ph** [comment]
I am fine with only Linux, @jlind23 We should probably exclude `dev-tools/*` WDYT?

**Comment by v1v** [comment]
> Having only the Linux coverage can be a good first step.

Thanks, https://github.com/elastic/elastic-agent/pull/377/commits/499e299bfd6a5fa2b0033f32b02edc8caa0488fc is the one only driving the test_coverage for linux

**Comment by jlind23** [comment]
@ph agree, let's remove it.

**Comment by v1v** [comment]
> @ph agree, let's remove it.

How? I could strip those entries in the generated cov file, but I'm not sure if the go coverage supports something natively, any hints?

**Comment by ph** [comment]
@v1v I've look at that, seems not a simple solution than removing the files or path from the coverage output https://stackoverflow.com/questions/50065448/how-to-ignore-generated-files-from-go-test-coverage

**Comment by v1v** [comment]
@ph, 
do you think we can move on with this PR as is? Then someone in your team can help with ignoring nay giving list of files from the code coverage report in a follow up?

**Comment by jlind23** [comment]
@v1v sure move ahead, we will update it afterwards.

**Comment by v1v** [comment]
I only need some 👍 in the code review, so I can merge it :)

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**